consultation_analyser/consultations/models.py

```diff
@@ -495,3 +496,168 @@ class EvidenceRichMapping(UUIDPrimaryKeyModel, TimeStampedModel):
+        if not self.slug:
+            base_slug = slugify(self.title)[:250]
+            slug = base_slug
+            counter = 1
+            while Consultation.objects.filter(slug=slug).exclude(pk=self.pk).exists():
+                slug = f"{base_slug}-{counter}"
+                counter += 1
+            self.slug = slug
+        return super().save(*args, **kwargs)
+
+    class Meta(UUIDPrimaryKeyModel.Meta, TimeStampedModel.Meta):
+        constraints = [
+            models.UniqueConstraint(fields=["slug"], name="unique_consultation_slug"),
+        ]
+
+
+class Question(UUIDPrimaryKeyModel, TimeStampedModel):
+    """
+    Combined question model - can have free text, multiple choice, or both.
+    Replaces the Question/QuestionPart split.
+    """
+
+    consultation = models.ForeignKey(Consultation, on_delete=models.CASCADE)
+    text = models.TextField()
+    slug = models.SlugField(max_length=256)
+    number = models.IntegerField()
+
+    # Question configuration
+    has_free_text = models.BooleanField(default=True)
+    has_multiple_choice = models.BooleanField(default=False)
+    multiple_choice_options = models.JSONField(
+        null=True, blank=True
+    )  # List of options when has_multiple_choice=True
+
+    def save(self, *args, **kwargs):
+        if not self.slug:
+            base_slug = slugify(self.text)[:240]
+            self.slug = f"{base_slug}-{self.number}" if base_slug else str(self.number)
+        return super().save(*args, **kwargs)
+
+    class Meta(UUIDPrimaryKeyModel.Meta, TimeStampedModel.Meta):
+        constraints = [
+            models.UniqueConstraint(fields=["consultation", "slug"], name="unique_question_slug"),
+            models.UniqueConstraint(
+                fields=["consultation", "number"], name="unique_question_number"
+            ),
+        ]
+        ordering = ["number"]
+        indexes = [
+            models.Index(fields=["consultation", "has_free_text"]),
+        ]
+
+
+class Respondent(UUIDPrimaryKeyModel, TimeStampedModel):
+    consultation = models.ForeignKey(Consultation, on_delete=models.CASCADE)
+    themefinder_id = models.IntegerField(null=True, blank=True)
+    demographics = models.JSONField(default=dict)
+
+    class Meta(UUIDPrimaryKeyModel.Meta, TimeStampedModel.Meta):
+        indexes = [
+            models.Index(fields=["consultation", "themefinder_id"]),
+        ]
+
+    @property
+    def identifier(self):
+        return self.themefinder_id if self.themefinder_id else self.id
+
+
+class Response(UUIDPrimaryKeyModel, TimeStampedModel):
+    """Response to a question - can include both free text and multiple choice"""
+
+    respondent = models.ForeignKey(Respondent, on_delete=models.CASCADE)
+    question = models.ForeignKey(Question, on_delete=models.CASCADE)
+
+    # Response content
+    free_text = models.TextField(blank=True)  # Free text response
+    chosen_options = models.JSONField(default=list)  # Multiple choice selections
```

Comment:
Again, could this be an ArrayField? 

Should we have some kind of validation to check that the chosen options are actually valid for the question (perhaps we could write a check on save if having a custom validator is too fiddly?)

Reply:
Or, should we be treating multiple choice options more like themes, with reference to a different table? Will users be filtering on multiple choice options?